src/main/java/org/zalando/nakadi/service/subscription/SubscriptionService.java

```diff
@@ -214,39 +217,54 @@ public Result listSubscriptions(final String owningApplication, final Set<String
 
         return eventTypes.stream()
                 .map(eventType -> {
-                    final Set<SubscriptionEventTypeStats.Partition> statPartitions = Arrays.stream(partitions)
-                            .filter(partition -> eventType.getTopic().equals(partition.getKey().getTopic()))
-                            .flatMap(partition ->
-                                    filterTopicPartitions(zkSubscriptionClient, topicPartitions, partition))
+                    final Set<SubscriptionEventTypeStats.Partition> statPartitions =
+                            topicPartitions.stream()
+                            .sorted(Comparator.comparing(topicPartition ->
+                                    Integer.valueOf(topicPartition.getPartitionId())))
+                            .filter(partition -> eventType.getTopic().equals(partition.getTopicId()))
+                            .map(Try.wrap(partition ->
+                                    mergePartitions(zkSubscriptionClient, zkSubscriptionNode, partition)))
+                            .map(Try::getOrThrow)
                             .collect(Collectors.toSet());
                     return new SubscriptionEventTypeStats(eventType.getName(), statPartitions);
                 })
                 .collect(Collectors.toList());
     }
 
-    private Stream<SubscriptionEventTypeStats.Partition> filterTopicPartitions(
+    private SubscriptionEventTypeStats.Partition mergePartitions(
             final ZkSubscriptionClient zkSubscriptionClient,
-            final List<TopicPartition> topicPartitions,
-            final Partition partition) {
-        return topicPartitions.stream()
-                .filter(topicPartition ->
-                        partition.getKey().getPartition().equals(topicPartition.getPartitionId()))
-                .map(Try.wrap(topicPartition ->
-                        createPartition(zkSubscriptionClient, partition, topicPartition)))
-                .map(Try::getOrThrow);
+            final ZkSubscriptionNode zkSubscriptionNode,
+            final TopicPartition topicPartition) throws NakadiException {
+        final boolean hasSessions = zkSubscriptionNode.getSessions().length > 0;
+
+        final Partition partition = Arrays.stream(zkSubscriptionNode.getPartitions())
+                .filter(p -> p.getKey().getPartition().equals(topicPartition.getPartitionId()))
+                .findFirst()
+                .orElse(null);
+
+        return createPartition(zkSubscriptionClient, partition, topicPartition, hasSessions);
     }
 
     private SubscriptionEventTypeStats.Partition createPartition(final ZkSubscriptionClient zkSubscriptionClient,
                                                                  final Partition partition,
```

Comment:
@v1ctor Yes, it will be null, should I mark it as @Nullable? I can not remember that rule.